applications/luci-app-acme/htdocs/luci-static/resources/view/acme.js

```diff
@@ -21,120 +31,455 @@ return view.extend({
+
+		o = _addDnsCredentialField(s, "dns_zilore", "Zilore_Key", "Zilore API Key", "");
+
+		o = _addDnsCredentialField(s, "dns_zone", "ZONE_Username", "Zone.ee Username", "");
+		o = _addDnsCredentialField(s, "dns_zone", "ZONE_Key", "Zone.ee API Key", "");
+
+		o = _addDnsCredentialField(s, "dns_zonomi", "ZM_Key", "Zonomi.com API Key", "");
+
+
+		o = s.taboption("challenge_dns", form.DynamicList, "credentials", _("DNS API credentials"),
 			_("The credentials for the DNS API mode selected above. " +
 				"See https://github.com/acmesh-official/acme.sh/wiki/dnsapi for the format of credentials required by each API. " +
 				"Add multiple entries here in KEY=VAL shell variable format to supply multiple credential variables."))
 		o.datatype = "list(string)";
 		o.depends("validation_method", "dns");
 		o.modalonly = true;
 
-		o = s.taboption('challenge', form.Value, "calias", _("Challenge Alias"),
+		o = s.taboption("challenge_dns", form.Value, "calias", _("Challenge Alias"),
 			_("The challenge alias to use for ALL domains. " +
 				"See https://github.com/acmesh-official/acme.sh/wiki/DNS-alias-mode for the details of this process. " +
 				"LUCI only supports one challenge alias per certificate."));
 		o.depends("validation_method", "dns");
 		o.modalonly = true;
 
-		o = s.taboption('challenge', form.Value, "dalias", _("Domain Alias"),
+		o = s.taboption("challenge_dns", form.Value, "dalias", _("Domain Alias"),
 			_("The domain alias to use for ALL domains. " +
 				"See https://github.com/acmesh-official/acme.sh/wiki/DNS-alias-mode for the details of this process. " +
 				"LUCI only supports one challenge domain per certificate."));
 		o.depends("validation_method", "dns");
 		o.modalonly = true;
 
+		o = s.taboption('advanced', form.Flag, "use_staging", _("Use staging server"),
+			_("Get certificate from the Letsencrypt staging server " +
+				"(use for testing; the certificate won't be valid)."));
+		o.modalonly = true;
+
+		o = s.taboption('advanced', form.ListValue, "key_type", _("Key size"),
+			_("Key size (and type) for the generated certificate."));
+		o.value("", "");
+		o.value("rsa2048", _("RSA 2048 bits"));
+		o.value("rsa3072", _("RSA 3072 bits"));
+		o.value("rsa4096", _("RSA 4096 bits"));
+		o.value("ec256", _("ECC 256 bits"));
+		o.value("ec384", _("ECC 384 bits"));
+		o.cfgvalue = function(section_id, set_value) {
+			var keylength = uci.get('acme', section_id, 'keylength');
+			if (keylength) {
+				// migrate old RSA to new keytype
+				switch (keylength) {
+					case "2048": return "rsa2048";
+					case "3072": return "rsa3072";
+					case "4096": return "rsa4096";
+					case "ec-256": return "ec256";
+					case "ec-384": return "ec384";
+					default: return ""; // bad value
+				}
+			}
+			return set_value;
+		};
+		o.write = function(section_id, value) {
+			// remove old keylength
+			uci.set('acme', section_id, 'keylength', '');
```

Comment:
You might need to check, but I think `uci.delete()` is better suited. Otherwise you might always write two values.

Reply:
I used uci.unset()